Common/src/geometry/CPhysicalGeometry.cpp

```diff
@@ -7456,6 +7456,92 @@ void CPhysicalGeometry::MatchPeriodic(CConfig        *config,
   delete [] Buffer_Recv_Vertex;
   delete [] Buffer_Recv_Marker;
+}
+
+void CPhysicalGeometry::FindUniqueNode_PeriodicBound(CConfig *config) {
+
+  /*-------------------------------------------------------------------------------------------*/
+  /*--- Find reference node on the 'inlet' streamwise periodic marker for the computation   ---*/
+  /*--- of recovered pressure/temperature, such that this found node is independent of the  ---*/
+  /*--- number of ranks. This does not affect the 'correctness' of the solution as the      ---*/
+  /*--- absolute value is arbitrary anyway, but it assures that the solution does not change---*/
+  /*--- with a higher number of ranks. If the periodic markers are a line\plane and the     ---*/
+  /*--- streamwise coordiante vector is perpendicular to that |--->|, the choice of the     ---*/
+  /*--- reference node is not relevant at all. This is probably true for most streamwise    ---*/
+  /*--- periodic cases. Other cases where it is relevant could look like this (--->( or     ---*/
+  /*--- \--->\ . The chosen metric is the minimal distance to the origin.                   ---*/
+  /*-------------------------------------------------------------------------------------------*/
+
+  /*--- Initialize/Allocate variables. ---*/
+  su2double min_norm = 0.0;
+
+  vector<su2double> Buffer_Send_RefNode(nDim, 1e300),
+                    Buffer_Recv_RefNode(static_cast<size_t>(size)*nDim);
+
+  /*-------------------------------------------------------------------------------------------*/
+  /*--- Step 1: Find a unique reference node on each rank and communicate them such that    ---*/
+  /*---         each process has the local ref-nodes from every process. Most processes     ---*/
+  /*---         won't have a boundary with the streamwise periodic 'inlet' marker,          ---*/
+  /*---         therefore the default value of the send value is set super high.            ---*/
+  /*-------------------------------------------------------------------------------------------*/
+
+  for (unsigned short iMarker = 0; iMarker < config->GetnMarker_All(); iMarker++) {
+    if (config->GetMarker_All_KindBC(iMarker) == PERIODIC_BOUNDARY) {
+
+      /*--- 1 is the receiver/'inlet', 2 is the donor/'outlet', 0 if no PBC at all. ---*/
+      auto iPeriodic = config->GetMarker_All_PerBound(iMarker);
+      if (iPeriodic == 1) {
+
+        for (auto iVertex = 0ul; iVertex < GetnVertex(iMarker); iVertex++) {
+
+          auto iPoint = vertex[iMarker][iVertex]->GetNode();
+
+          /*--- Get the squared norm of the current point. ---*/
+          auto norm = GeometryToolbox::SquaredNorm(nDim, nodes->GetCoord(iPoint));
+
+          /*--- Check if new unique reference node is found. ---*/
+          if (norm < min_norm || iVertex == 0) {
+            min_norm = norm;
+            for (unsigned short iDim = 0; iDim < nDim; iDim++)
+              Buffer_Send_RefNode[iDim] = nodes->GetCoord(iPoint,iDim);
+          }
```

Comment:
I would initialize norm to a large value to avoid the iVertex logic, and make Buffer_Send a const pointer, i.e. without copying the data.

Reply:
I have to admit I do not understand  the `make Buffer_Send a const pointer, i.e. without copying the data.` part. I now changed the code to communicate the dist as well as you suggested, so Buffer_Send  has size nDim+1.
But for now assume Buffer_Send  just contains the coordinates then I could:
```c++
//outside
const su2double *Buffer_Send_RefNode;
...
//in the loop
Buffer_Send_RefNode = nodes->GetCoord(iPoint);
```
because I dont want the data Buffer_Send_RefNode  points to to change. That is ok, but if i wanna include the distance into that Buffer_Send_RefNode I need to have dedicated memory for that. Maybe my brain is in a const state unable to learn or you had two different ideas which contradict each other to some degree :)

Reply:
I would initialize *min_* norm to a large value to avoid the iVertex logic ... completely right, that looks a bit nicer now

Reply:
They were indeed 2+ ideias, but not mutually exclusive, you can determine "who" is the closest point inside the loop by keeping either the corresponding "iPoint" or directly a const ptr to the closest coordinates, then you copy the coordinates into the send buffer only once outside of the loop.

Keeping "iPointMin" is actually the "right" way to do it in the sense that you don't rely on the xyz coordinates being together in memory (which the pointer implies), and with that you separate the algorithm (find min) from the data structure (how the coordinates are organized in memory).
This separation is the main reason I was talking about that "matrix view" for gradients, and it is why (according to the authors) the algorithms in the STL (the template library, not the format) take iterators instead of pointers.

Reply:
I chose the "right" way ;) ... at least that is what i hope

Reply:
I hope it is too...